integration_tests/cairo0_test.go

```diff
@@ -0,0 +1,318 @@
+		t.Logf("trace:\n%s\n", traceRepr(trace))
+		writeToFile(path)
+	}
+	if !assert.Equal(t, pyMemory, memory) {
+		t.Logf("pymemory;\n%s\n", memoryRepr(pyMemory))
+		t.Logf("memory;\n%s\n", memoryRepr(memory))
+		writeToFile(path)
+	}
+}
+
+var zerobench = flag.Bool("zerobench", false, "run integration tests and generate benchmarks file")
+
+func TestCairoZeroFiles(t *testing.T) {
+	file, err := os.OpenFile(whitelistFile, os.O_CREATE|os.O_WRONLY|os.O_TRUNC, 0644)
+	if err != nil {
+		panic(fmt.Errorf("failed to open file: %w", err))
+	}
+
+	file.Close()
+
+	roots := []string{
+		"./cairo_zero_hint_tests/",
+		"./cairo_zero_file_tests/",
+		"./builtin_tests/",
+	}
+
+	// filter is for debugging purposes
+	filter := Filter{}
+	filter.init()
+
+	benchmarkMap := make(map[string][]int)
+
+	sem := make(chan struct{}, 5) // semaphore to limit concurrency
+	var wg sync.WaitGroup         // WaitGroup to wait for all goroutines to finish
+
+	for _, root := range roots {
+		testFiles, err := os.ReadDir(root)
+		require.NoError(t, err)
+
+		for _, dirEntry := range testFiles {
+			if dirEntry.IsDir() || isGeneratedFile(dirEntry.Name()) {
+				continue
+			}
+
+			name := dirEntry.Name()
+			path := filepath.Join(root, name)
+
+			errorExpected := false
+			if name == "range_check__small.cairo" {
+				errorExpected = true
+			}
+			if !filter.filtered(name) {
+				continue
+			}
+			// we run tests concurrently if we don't need benchmarks
+			if !*zerobench {
+				sem <- struct{}{} // acquire a semaphore slot
+				wg.Add(1)
+
+				go func(path, name string) {
+					defer wg.Done()
+					defer func() { <-sem }() // release the semaphore slot when done
+					runAndTestCairoZeroFile(t, path, name, benchmarkMap, *zerobench, errorExpected)
+				}(path, name)
+			} else {
+				runAndTestCairoZeroFile(t, path, name, benchmarkMap, *zerobench, errorExpected)
+			}
+		}
+	}
+
+	wg.Wait() // wait for all goroutines to finish
+
+	for _, root := range roots {
+		clean(root)
+	}
+
+	if *zerobench {
+		WriteBenchMarksToFile(benchmarkMap)
+	}
+}
+
+// given a path to a cairo zero file, it compiles it
+// and return the compilation path
+func compileCairoZeroCode(path string) (string, error) {
+	if filepath.Ext(path) != ".cairo" {
+		return "", fmt.Errorf("compiling non cairo file: %s", path)
+	}
+	compiledOutput := swapExtenstion(path, compiledSuffix)
+
+	var cliCommand string
+	var args []string
+
+	cliCommand = "cairo-compile"
+	args = []string{
+		path,
+		"--proof_mode",
+		"--no_debug_info",
+		"--output",
+		compiledOutput,
+	}
+	cmd := exec.Command(cliCommand, args...)
+
+	res, err := cmd.CombinedOutput()
+	if err != nil {
+		return "", fmt.Errorf(
+			"%s %s: %w\n%s", cliCommand, path, err, string(res),
+		)
+	}
+
+	return compiledOutput, nil
+}
+
+// given a path to a compiled cairo zero file, execute it using the
+// python vm and returns the trace and memory files location
+func runPythonVmZero(path, layout string) (time.Duration, string, string, error) {
```

Comment:
Isn't the Python VM only used for Cairo zero? You could just call this `runPythonVM` instead

Reply:
you are right, renamed